wp-resources/plugins/sil-dictionary-webonary/webonary/Webonary_Cloud.php

```diff
@@ -30,9 +31,26 @@ private static function remoteGetJson($path, $apiParams = array()) {
         $encoded_path = array_map('rawurlencode', explode('/', $path));
         $url = rtrim(WEBONARY_CLOUD_API_URL, '/') . '/' . implode('/', $encoded_path);
 
-        if (count($apiParams)) {
-            $url .= '?' . build_query(array_map('urlencode', $apiParams));
-        }
+		if (count($apiParams)) {
+
+			// expanded this to allow for an array of 'parts of speech' to be passed as a value
+			$pieces = [];
+			foreach ($apiParams AS $key => $val) {
+
+				if (is_array($val)) {
+
+					// send each value in the array separately, using the same key
+					foreach($val AS $v) {
+						$pieces[] = $key . '=' . urlencode($v);
+					}
+				}
+				else {
+					$pieces[] = $key . '=' . urlencode($val);
+				}
+			}
+
+			$url .= '?' . implode('&', $pieces);
+		}
 
```

Comment:
Line 60 below is now causing problems, perhaps because of PHP upgrade?

```
        if (WP_DEBUG_WEBONARY_CLOUD){
            error_log('Getting results from ' . $url);
        }
```
Should be
```
        if (defined('WP_DEBUG_WEBONARY_CLOUD') && WP_DEBUG_WEBONARY_CLOUD){
            error_log('Getting results from ' . $url);
        }
```

Reply:
@hanatgit the constant `WP_DEBUG_WEBONARY_CLOUD` is defined (optionally) in the `wp-config.php` file.